frontend/src/app/main/data/workspace/libraries.cljs

```diff
@@ -625,17 +664,22 @@
                                (->> file-changes (remove nil?) (map second) (flatten)))]
         (log/debug :msg "SYNC-FILE finished" :js/rchanges (log-changes
                                                             rchanges
-                                                            local-library))
+                                                            file))
         (rx/concat
           (rx/of (dm/hide-tag :sync-dialog))
           (when rchanges
-            (rx/of (dwc/commit-changes rchanges uchanges {:commit-local? true})))
-          (when (not= file-id (:current-file-id state))
-            (rp/mutation :update-sync
-                         {:file-id (get-in state [:workspace-file :id])
-                          :library-id file-id}))
+            (rx/of (dwc/commit-changes rchanges uchanges {:commit-local? true
+                                                          :file-id file-id})))
+          (when (not= file-id library-id)
+            ;; When we have just updated the library file, give some time for the
+            ;; update to finish, before marking this file as synced.
+            ;; TODO: look for a more precise way of syncing this.
+            (rx/concat (rx/timer 3000)
```

Comment:
maybe this timeout can be avoided using the `stream` (second argument passed to `watch`) for wait the corresponding `changes-commited` and then proced with the `:update-sync` mutation.

Reply:
This will require more careful thinking, and considering side cases. I take note to do it ASAP.